src/widget/findonwebmenufactory.cpp

```diff
@@ -0,0 +1,26 @@
+#include "findonwebmenufactory.h"
+
+#include <QMenu>
+#include <QtDebug>
+
+#include "findondiscogsmenu.h"
+#include "findonlastfmmenu.h"
+#include "findonsoundcloudmenu.h"
+#include "track/track.h"
+#include "wfindonwebmenu.h"
+
+void FindOnWebMenuFactory::createFindOnWebSubmenus(QMenu* pFindOnWebMenu,
+        const Track& track,
+        bool isSoundcloudEnabled,
+        bool isDiscogsEnabled,
+        bool isLastfmEnabled) {
+    if (isSoundcloudEnabled) {
+        new FindOnSoundcloudMenu(pFindOnWebMenu, track);
```

Comment:
we should avoid plain new. Can we use her make_parented?

Reply:
Yeap, I have tried these:

`auto pSoundCloudMenu = make_parented<WFindOnWebMenu>(new FindOnSoundcloudMenu(pFindOnWebMenu, track));`

`auto pFindOnDiscogsMenu = make_parented<QMenu>(new FindOnDiscogsMenu(pFindOnWebMenu, track));`

They both worked, Which one I should use?

But I'm still not sure which one owns the Service Menus, WFindOnWebMenu or QMenu? Also WFindOnWebMenu inherits from QMenu, so making parented QMenu and WFindOnWebMenu is the same thing? 
Or 
Making QMenu deletes the object after the main menu "WTrackMenu" closed
Making WFindOnWebMenu deletes the object after the FindOnWeb menu is closed (WTrackMenu is still open in that case)